web/caret_browsing.js

```diff
@@ -23,10 +23,20 @@ class CaretBrowsingMode {
 
   #viewerContainer;
 
-  constructor(mainContainer, viewerContainer, toolbarContainer) {
+  constructor(eventBus, mainContainer, viewerContainer, toolbarContainer) {
     this.#mainContainer = mainContainer;
     this.#viewerContainer = viewerContainer;
-    this.#toolBarHeight = toolbarContainer?.getBoundingClientRect().height ?? 0;
+
+    const computeToolBarHeight = () => {
+      this.#toolBarHeight =
+        toolbarContainer?.getBoundingClientRect().height ?? 0;
+    };
+    computeToolBarHeight();
+
+    eventBus._on("toolbardensity", async () => {
+      await Promise.resolve(); // Ensure that the UI has time to update first.
```

Comment:
Is it enough ?
Maybe `setTimeout(..., 0)` will give us slightly more time. 
Normally, the user is changing the toolbar height in using toolbar config ui or from about:config, in both cases `setTimeout` should be enough.
Maybe I'm a bit overthinking here.

Reply:
Mainly I just wanted to ensure that https://github.com/mozilla/pdf.js/blob/3cdc3258d94a06af658e50290bf709426adf44b8/web/toolbar.js#L129-L140 has run, but I suppose that using `setTimeout` is indeed safer here.

Reply:
An other possibility would be to add a ResizeObserver on the toolbar.

Reply:
Sure, please see the updated patch.